src/Documents/BulkInsert/BulkInsertWriterBase.ts

```diff
@@ -51,8 +51,12 @@ export class BulkInsertWriterBase implements IDisposable {
         this.onCurrentWriteStreamSet(this._currentWriter);
     }
 
+    public isFlushNeeded() {
+        return this._currentWriter.length > this._maxSizeInBuffer;
+    }
+
     public async flushIfNeeded(force = false): Promise<void> {
-        if (this._currentWriter.length > this._maxSizeInBuffer || this._asyncWriteDone) {
+        if (this.isFlushNeeded() || this._asyncWriteDone) {
```

Comment:
I think that `this._asyncWriteDone` should be in `isFlushNeeded`

Reply:
yes. It generates more falses in tryStoreSync, but we wait less overall.